examples/DDIM/train.jl

```diff
@@ -0,0 +1,177 @@
+using Lux
+using Random
+using Images
+using ImageFiltering
+using Interpolations
+using MLUtils
+using Optimisers
+using Statistics
+using ProgressBars
+using Zygote
+using CUDA
+using BSON
+using Comonicon
+using Printf
+
+include("./model.jl")
+
+# Image dataset
+# `image_files` are vector of image file path
+# if `use_cache=true`, preprocessed image is cached on memory
+struct ImageDataset
+    image_files::Vector{AbstractString}
+    preprocess::Function
+    use_cache::Bool
+    cache::Vector{Union{Nothing, AbstractArray{Float32, 3}}}
+end
+
+function ImageDataset(dirpath::AbstractString, preprocess::Function, use_cache::Bool)
+    image_files = joinpath.(dirpath, readdir(dirpath))
+    cache = map(x -> nothing, image_files)
+    return ImageDataset(image_files, preprocess, use_cache, cache)
+end
+
+Base.length(ds::ImageDataset) = length(ds.image_files)
+
+function Base.getindex(ds::ImageDataset, i::Int)
+    if ds.use_cache && !isnothing(ds.cache[i])
+        return ds.cache[i]
+    else
+        img = Images.load(ds.image_files[i])
+        img = ds.preprocess(img)
+        img = permutedims(channelview(img), (2, 3, 1))
+        if ds.use_cache
+            ds.cache[i] = img
+        end
+        return Float32.(img)
+    end
+end
+
+function center_crop(image::Matrix{RGB{T}}) where {T <: Real}
```

Comment:
Use `DataAugmentation` or `Augmentor` for these. Don't reimplement those functionality

Reply:
Thanks. I didn't know those packages.
I've used `Augmentor`